internal/builders/controlplane/deployment.go

```diff
@@ -161,115 +191,129 @@ func (d *Deployment) buildPKIVolume(podSpec *corev1.PodSpec, tcp *kamajiv1alpha1
+	found, index := utilities.HasNamedVolume(podSpec.Volumes, usrLocalShareCaCertificateVolumeName)
+	if !found {
+		index = len(podSpec.Volumes)
 		podSpec.Volumes = append(podSpec.Volumes, corev1.Volume{})
 	}
 
-	podSpec.Volumes[usrLocalShareCACertificates] = corev1.Volume{
-		Name: "usr-local-share-ca-certificates",
-		VolumeSource: corev1.VolumeSource{
-			Secret: &corev1.SecretVolumeSource{
-				SecretName:  tcp.Status.Certificates.CA.SecretName,
-				DefaultMode: pointer.Int32(420),
-			},
+	podSpec.Volumes[index].Name = usrLocalShareCaCertificateVolumeName
+	podSpec.Volumes[index].VolumeSource = corev1.VolumeSource{
+		Secret: &corev1.SecretVolumeSource{
+			SecretName:  tcp.Status.Certificates.CA.SecretName,
+			DefaultMode: pointer.Int32(420),
 		},
 	}
 }
 
 func (d *Deployment) buildSchedulerVolume(podSpec *corev1.PodSpec, tcp *kamajiv1alpha1.TenantControlPlane) {
-	if index := int(schedulerKubeconfig) + 1; len(podSpec.Volumes) < index {
+	found, index := utilities.HasNamedVolume(podSpec.Volumes, schedulerKubeconfigVolumeName)
+	if !found {
+		index = len(podSpec.Volumes)
 		podSpec.Volumes = append(podSpec.Volumes, corev1.Volume{})
 	}
 
-	podSpec.Volumes[schedulerKubeconfig] = corev1.Volume{
-		Name: "scheduler-kubeconfig",
-		VolumeSource: corev1.VolumeSource{
-			Secret: &corev1.SecretVolumeSource{
-				SecretName:  tcp.Status.KubeConfig.Scheduler.SecretName,
-				DefaultMode: pointer.Int32(420),
-			},
+	podSpec.Volumes[index].Name = schedulerKubeconfigVolumeName
+	podSpec.Volumes[index].VolumeSource = corev1.VolumeSource{
+		Secret: &corev1.SecretVolumeSource{
+			SecretName:  tcp.Status.KubeConfig.Scheduler.SecretName,
+			DefaultMode: pointer.Int32(420),
 		},
 	}
 }
 
 func (d *Deployment) buildControllerManagerVolume(podSpec *corev1.PodSpec, tcp *kamajiv1alpha1.TenantControlPlane) {
-	if index := int(controllerManagerKubeconfig) + 1; len(podSpec.Volumes) < index {
+	found, index := utilities.HasNamedVolume(podSpec.Volumes, controllerManagerKubeconfigVolumeName)
+	if !found {
+		index = len(podSpec.Volumes)
 		podSpec.Volumes = append(podSpec.Volumes, corev1.Volume{})
 	}
 
-	podSpec.Volumes[controllerManagerKubeconfig] = corev1.Volume{
-		Name: "controller-manager-kubeconfig",
-		VolumeSource: corev1.VolumeSource{
-			Secret: &corev1.SecretVolumeSource{
-				SecretName:  tcp.Status.KubeConfig.ControllerManager.SecretName,
-				DefaultMode: pointer.Int32(420),
-			},
+	podSpec.Volumes[index].Name = controllerManagerKubeconfigVolumeName
+	podSpec.Volumes[index].VolumeSource = corev1.VolumeSource{
+		Secret: &corev1.SecretVolumeSource{
+			SecretName:  tcp.Status.KubeConfig.ControllerManager.SecretName,
+			DefaultMode: pointer.Int32(420),
 		},
 	}
 }
 
+// SetAdditionalVolumes allows from user-space to add additional volumes:
+// it reset the podSpec.volumes key, the following modifiers are ensuring the idempotency of the builder.
+// Thus, this function must be called before SetVolumes.
```

Comment:
```suggestion
// SetAdditionalVolumes must be called before SetVolumes: the user-space ones are going to be prepended
// to simplify the management of the Kamaji ones during the create or update action.


// SetAdditionalVolumes allows from user-space to add additional volumes:
// it reset the podSpec.volumes key, the following modifiers are ensuring the idempotency of the builder.
// Thus, this function must be called before SetVolumes.
```
I was thinking we can be consistent with the description of `SetAdditionalContainers`, what do you think?